Interactive rebase with a break prevents lazygit from startup

**Describe the bug**
Lazygit chokes on an interactive rebase with a break in the todo list.

**To Reproduce**
Steps to reproduce the behavior:
1. Start interactive rebase with `git rebase -i main`
2. Add two breaks anywhere
3. Start the rebase
4. Start lazygit - will crash immediately.

**Expected behavior**
No crash!

**Stacktrace**
```
panic: runtime error: slice bounds out of range [2:1]

goroutine 83 [running]:
github.com/jesseduffield/lazygit/pkg/commands/loaders.(*CommitLoader).getInteractiveRebasingCommits(0xc0000bf590)
	/tmp/makepkg/lazygit-git/src/src/github.com/jesseduffield/lazygit/pkg/commands/loaders/commits.go:320 +0x44b
github.com/jesseduffield/lazygit/pkg/commands/loaders.(*CommitLoader).getRebasingCommits(0xbdcc28, 0xc000202120)
	/tmp/makepkg/lazygit-git/src/src/github.com/jesseduffield/lazygit/pkg/commands/loaders/commits.go:244 +0x4f
github.com/jesseduffield/lazygit/pkg/commands/loaders.(*CommitLoader).getHydratedRebasingCommits(0xc0000bf590, 0xaeb365)
	/tmp/makepkg/lazygit-git/src/src/github.com/jesseduffield/lazygit/pkg/commands/loaders/commits.go:194 +0x33
github.com/jesseduffield/lazygit/pkg/commands/loaders.(*CommitLoader).MergeRebasingCommits(0xc0000bf590, {0x109ee90, 0x0, 0x0})
	/tmp/makepkg/lazygit-git/src/src/github.com/jesseduffield/lazygit/pkg/commands/loaders/commits.go:145 +0x14b
github.com/jesseduffield/lazygit/pkg/commands/loaders.(*CommitLoader).GetCommits(0xc0000bf590, {0x1, {0x0, 0x0}, 0x1, {0xaeb551, 0x4}, 0x0})
	/tmp/makepkg/lazygit-git/src/src/github.com/jesseduffield/lazygit/pkg/commands/loaders/commits.go:78 +0xdf
github.com/jesseduffield/lazygit/pkg/gui.(*Gui).refreshCommitsWithLimit(0xc00000c3c0)
	/tmp/makepkg/lazygit-git/src/src/github.com/jesseduffield/lazygit/pkg/gui/commits_panel.go:119 +0x1a5
github.com/jesseduffield/lazygit/pkg/gui.(*Gui).refreshCommits.func2()
	/tmp/makepkg/lazygit-git/src/src/github.com/jesseduffield/lazygit/pkg/gui/commits_panel.go:94 +0x2f
github.com/jesseduffield/lazygit/pkg/utils.Safe.func1()
	/tmp/makepkg/lazygit-git/src/src/github.com/jesseduffield/lazygit/pkg/utils/utils.go:95 +0x1b
github.com/jesseduffield/lazygit/pkg/utils.SafeWithError(0x0)
	/tmp/makepkg/lazygit-git/src/src/github.com/jesseduffield/lazygit/pkg/utils/utils.go:106 +0x67
github.com/jesseduffield/lazygit/pkg/utils.Safe(0x0)
	/tmp/makepkg/lazygit-git/src/src/github.com/jesseduffield/lazygit/pkg/utils/utils.go:95 +0x35
created by github.com/jesseduffield/lazygit/pkg/gui.(*Gui).refreshCommits
	/tmp/makepkg/lazygit-git/src/src/github.com/jesseduffield/lazygit/pkg/gui/commits_panel.go:93 +0x167
```

**Desktop:**
 - OS: Arch Linux
 - Lazygit Version: commit=d13a6481, build date=20220128.090800, build source=unknown, version=v0.32.1.d13a6481, os=linux, arch=amd64

**Additional context**
The same thing happens when running lazygit and then doing the same rebase on a different terminal. Lazygit will then crash on the next refresh.

--- a/go.mod
+++ b/go.mod
@@ -9,6 +9,7 @@ require (
 	github.com/cli/safeexec v1.0.0
 	github.com/cloudfoundry/jibber_jabber v0.0.0-20151120183258-bcc4c8345a21
 	github.com/creack/pty v1.1.11
+	github.com/fsmiamoto/git-todo-parser v0.0.2
 	github.com/fsnotify/fsnotify v1.4.7
 	github.com/go-errors/errors v1.4.2
 	github.com/gookit/color v1.4.2

--- a/pkg/commands/loaders/commits.go
+++ b/pkg/commands/loaders/commits.go
@@ -1,6 +1,7 @@
 package loaders
 
 import (
+	"bytes"
 	"fmt"
 	"io/ioutil"
 	"os"
@@ -9,6 +10,7 @@ import (
 	"strconv"
 	"strings"
 
+	"github.com/fsmiamoto/git-todo-parser/todo"
 	"github.com/jesseduffield/generics/slices"
 	"github.com/jesseduffield/lazygit/pkg/commands/models"
 	"github.com/jesseduffield/lazygit/pkg/commands/oscommands"
@@ -307,21 +309,23 @@ func (self *CommitLoader) getInteractiveRebasingCommits() ([]*models.Commit, err
 	}
 
 	commits := []*models.Commit{}
-	lines := strings.Split(string(bytesContent), "\n")
 
-	for _, line := range lines {
-		if line == "" || line == "noop" {
-			return commits, nil
-		}
-		if strings.HasPrefix(line, "#") {
+	todos, err := todo.Parse(bytes.NewBuffer(bytesContent))
+	if err != nil {
+		self.Log.Error(fmt.Sprintf("error occurred while parsing git-rebase-todo file: %s", err.Error()))
+		return nil, nil
+	}
+
+	for _, t := range todos {
+		if t.Commit == "" {
+			// Command does not have a commit associated, skip
 			continue
 		}
-		splitLine := strings.Split(line, " ")
 		commits = slices.Prepend(commits, &models.Commit{
-			Sha:    splitLine[1],
-			Name:   strings.Join(splitLine[2:], " "),
+			Sha:    t.Commit,
+			Name:   t.Msg,
 			Status: "rebasing",
-			Action: splitLine[0],
+			Action: t.Command.String(),
 		})
 	}
 

